test/blackbox-tests/test-cases/pkg/opam-package-with-doc.t

```diff
@@ -0,0 +1,35 @@
+We test how a package using the `with-doc` variable behaves.
+
+  $ . ./helpers.sh
+  $ mkrepo
+  $ mkpkg  
+
+  $ mkpkg with-doc <<EOF
+  > depends: [ "doc-dep" {with-doc} ]
+  > build: [ "echo" "Building with-doc" {with-doc} ]
+  > EOF
+  $ mkpkg doc-dep <<EOF
+  > build: [ "echo" "Building doc-dep" ]
+  > EOF
+
+The doc dependency is currently ignored. We also make sure to translate the filter.
+  $ solve_translate_opam_filters with-doc
+  Solution for dune.lock:
+  - with-doc.0.0.1
+
+The filter is translated correctly.
+  $ cat dune.lock/with-doc.pkg 
+  (version 0.0.1)
+  
+  (build
+   (run
+    echo
+    (when %{pkg-self:with-doc} "Building with-doc")))
```

Comment:
This is an issue with our lock generation. This `when` should not be included b/c we evaluated `with-doc` to false when solving.

Reply:
This is with the `--experimental-filter-translation` option. When that is not set the filter is removed but the step stays.